prime-router/src/main/kotlin/ActionDetail.kt

```diff
@@ -0,0 +1,66 @@
+package gov.cdc.prime.router
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.module.kotlin.jacksonMapperBuilder
+import gov.cdc.prime.router.azure.db.tables.pojos.Action
+import java.util.UUID
+
+private val mapper = jacksonMapperBuilder().build()
+/**
+ * @property scope of the result detail
+ * @property id of the result (depends on scope)
+ * @property details of the result
+ * @property row of csv related to message (set to -1 if not applicable)
+ */
+data class ActionDetail(
+    val scope: DetailScope,
+    val trackingId: String,
+    val responseMessage: ResponseMessage,
+    val row: Int = -1,
+    val context: JsonNode = mapper.valueToTree(responseMessage),
+    var reportId: UUID? = null,
+    var action: Action? = null,
```

Comment:
This shouldn't be nullable, a detail should always have an action associated with it.